src/client-side-encryption/clientEncryption.ts

```diff
@@ -0,0 +1,1015 @@
+import type {
+  ExplicitEncryptionContextOptions,
+  MongoCrypt,
+  MongoCryptOptions
+} from 'mongodb-client-encryption';
+
+import { type Binary, type Document, type Long, serialize } from '../bson';
+import { type BulkWriteResult } from '../bulk/common';
+import { type ProxyOptions } from '../cmap/connection';
+import { type Collection } from '../collection';
+import { type FindCursor } from '../cursor/find_cursor';
+import { type Db } from '../db';
+import { getMongoDBClientEncryption } from '../deps';
+import { type MongoClient } from '../mongo_client';
+import { type Filter } from '../mongo_types';
+import { type CreateCollectionOptions } from '../operations/create_collection';
+import { type DeleteResult } from '../operations/delete';
+import { type Callback, MongoDBCollectionNamespace } from '../utils';
+import { maybeCallback, promiseOrCallback } from './common';
+import * as cryptoCallbacks from './cryptoCallbacks';
+import { MongoCryptCreateDataKeyError, MongoCryptCreateEncryptedCollectionError } from './errors';
+import { type KMSProvider, type KMSProviders, loadCredentials } from './providers/index';
+import { type CSFLEKMSTlsOptions, StateMachine, type StateMachineExecutable } from './stateMachine';
+
```

Comment:
Types in this file were directly copied in from https://github.com/mongodb/libmongocrypt/blob/8588825112a93145bf82b4b7519070c101c45a43/bindings/node/index.d.ts#L448, with the exception of a few modifications:

- TLS options 
-